oak_functions/loader/src/server.rs

```diff
@@ -672,23 +607,16 @@ impl WasmHandler {
     }
 
     fn init_wasm_state(&self, request_bytes: Vec<u8>) -> anyhow::Result<WasmState> {
-        let mut extensions_indices = HashMap::new();
-        let mut extensions_metadata = HashMap::new();
+        let mut extensions = HashMap::new();
 
-        for (ind, factory) in self.extension_factories.iter().enumerate() {
+        // Create an extension from every factory.
+        for factory in self.extension_factories.iter() {
             let extension = factory.create()?;
-            let (name, signature) = extension.get_metadata();
-            extensions_indices.insert(ind + EXTENSION_INDEX_OFFSET, extension);
-            extensions_metadata.insert(name, (ind + EXTENSION_INDEX_OFFSET, signature));
+            extensions.insert(extension.get_handle(), extension);
         }
```

Comment:
Maybe this could be something like this, but not sure it's worth it, up to you.

```rust
let extensions = self
  .extension_factories
  .iter()
  .map(|f| f.create())
  .collect::<Result<Vec<_>, >()?
  .into_iter()
  .map(|e| (e.get_handle(), e))
  .collect::<HashMap<_, _>();
```

Or maybe something even nicer with https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.try_collect

Reply:
Next time I try :)